test/unit/test-adapter.js

```diff
@@ -146,12 +118,104 @@ describe('SlackMessageAdapter', function () {
 
   describe('#expressMiddleware()', function () {
     beforeEach(function () {
-      this.adapter = new SlackMessageAdapter(workingVerificationToken);
+      this.adapter = new SlackMessageAdapter(workingSigningSecret);
+      this.next = sinon.stub();
+      this.dispatch = sinon.stub();
+      this.res = sinon.stub({
+        setHeader: function () { },
+        end: function () { }
+      });
+      this.errFn = sinon.stub();
     });
+
     it('should return a function', function () {
       var middleware = this.adapter.expressMiddleware();
       assert.isFunction(middleware);
     });
+    it('should error when body parser is used', function () {
+      var middleware = this.adapter.expressMiddleware();
+      var req = { body: { } };
+      var res = this.res;
+      var next = this.next;
+      next.callsFake(function (err) {
+        assert.equal(err.code, errorCodes.BODY_PARSER_NOT_PERMITTED);
+      });
+      middleware(req, res, next);
+    });
+    it('should succeed with basic requests', function (done) {
+      var ts = Math.floor(Date.now() / 1000);
+      var signature = createRequestSignature(workingSigningSecret, ts, workingRawBody);
+      var adapter = this.adapter;
+      var middleware = adapter.expressMiddleware();
+      var dispatch = this.dispatch;
+      var res = this.res;
+      var next = this.next;
+      adapter.dispatch = dispatch;
+      // Create streamed request
+      const fakeRequest = new Readable({
+        read() {
+          this.push(workingRawBody);
+          this.push(null);
+        }
+      });
+      fakeRequest.headers = {
+        'x-slack-signature': signature,
+        'x-slack-request-timestamp': ts,
+        'content-type': contentType
```

Comment:
this can prob be a constant rather than a variable if the content-type is never changing (always urlencoded, never JSON).